File: PXWeb/Code/CustomControls/UserManualScreenReader.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Web;

namespace PXWeb.CustomControls
{
    public class UserManualScreenReader : System.Web.UI.WebControls.Literal
    {
        public string manualFor
        {
            get;
            set;
        }

        public string headerCode
        {
            get;
            set;
        }
        public string textCode
        {
            get;
            set;
        }

        public UserManualScreenReader() : base()
        {
            this.Text = "<span>Hello</span>";
            Load += Page_Load;
        }
        protected void Page_Load(object sender, EventArgs e)
        {
            Localize();
        }
        private void Localize()
        {
            string headerKey = "";
            string textKey = "";
            if (string.IsNullOrEmpty(manualFor)) {
                headerKey = headerCode; 
                textKey = textCode; 
            }
            else
            {
                headerKey = String.Format("UserManualScreenReader_{0}_Header", manualFor);
                textKey = String.Format("UserManualScreenReader_{0}_Text", manualFor);
            }

            string heading = PCAxis.Web.Core.Management.LocalizationManager.GetLocalizedString(headerKey);
            string longText = PCAxis.Web.Core.Management.LocalizationManager.GetLocalizedString(textKey);

            StringBuilder builder = new StringBuilder();
            builder.Append(@"<section aria-label=""");
            builder.Append(this.Page.Server.HtmlEncode(heading));
            builder.Append(@"""><span class=""screenreader-only"">");
            builder.Append(this.Page.Server.HtmlEncode(longText));
            builder.Append("</span></section>");
            
            this.Text = builder.ToString();
        }

    }
}

File: PCAxis.Web.Controls/UserManualScreenReader.vb
Public Class UserManualScreenReader
    Inherits System.Web.UI.WebControls.Literal

    Private _manualFor As String
    Private _headerCode As String
    Private _textCode As String

    Public Property manualForm() As String
        Get
            Return _manualFor
        End Get
        Set(ByVal value As String)
            _manualFor = value
        End Set
    End Property

    Public Property headerCode() As String
        Get
            Return _headerCode
        End Get
        Set(ByVal value As String)
            _headerCode = value
        End Set
    End Property

    Public Property textCode() As String
        Get
            Return _textCode
        End Get
        Set(ByVal value As String)
            _textCode = value
        End Set
    End Property

    Public Sub New()
        MyBase.New()
        Me.Text = "<span>Hello</span>"


    End Sub

    Protected Sub The_Load() Handles Me.Load
        Localize()
    End Sub


    Private Sub Localize()
        Dim headerKey As String
        Dim textKey As String

        If _manualFor Is Nothing Then
            headerKey = _headerCode
            textKey = _textCode
        Else
            headerKey = String.Format("UserManualScreenReader_{0}_Header", _manualFor)
            textKey = String.Format("UserManualScreenReader_{0}_Text", _manualFor)
        End If

        Dim heading = PCAxis.Web.Core.Management.LocalizationManager.GetLocalizedString(headerKey)
        Dim longText = PCAxis.Web.Core.Management.LocalizationManager.GetLocalizedString(textKey)

        Dim builder As New System.Text.StringBuilder
        builder.Append("<section aria-label=""")

        builder.Append(Page.Server.HtmlEncode(heading))
        builder.Append("""><span class=""screenreader-only"">")
        builder.Append(Page.Server.HtmlEncode(longText))
        builder.Append("</span></section>")
        Me.Text = builder.ToString
    End Sub

End Class
